lib/geo/utils.ex

```diff
@@ -1,6 +1,31 @@
 defmodule Geo.Utils do
   @moduledoc false
 
+  @spec string_to_float(String.t()) :: {:ok, float()} | {:error, term}
+  def string_to_float(str) when is_binary(str) do
+    case Float.parse(str) do
+      :error ->
+        {:error, :bad_arg}
+
+      {flt, ""} ->
+        {:ok, flt}
+
+      {_flt, _rest} ->
+        {:error, :bad_arg}
+    end
+  end
+
+  @spec string_to_float(String.t()) :: float()
```

Comment:
```suggestion
  @spec string_to_float!(String.t()) :: float() | no_return()
```

Reply:
huh, TIL about no_return, I'll add the test shortly